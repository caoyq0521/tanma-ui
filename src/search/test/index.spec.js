import { mount } from '@vue/test-utils';
import TmSearch from '../search.vue';

describe('tmSearch', () => {
  it('render tmSearch', () => {
    const wrapper = mount(TmSearch);
    expect(wrapper).toMatchSnapshot();
  });

  it('test placeholder', () => {
    const wrapper = mount(TmSearch, {
      propsData: {
        placeholder: '测试占位符内容',
      },
    });
    expect(wrapper.find('.tm-search input').attributes('placeholder')).toBe('测试占位符内容');
  });

  it('test maxlength', () => {
    const wrapper = mount(TmSearch, {
      propsData: {
        maxlength: 10,
        value: '-'.repeat(20),
      },
    });
    expect(wrapper.vm.$el.querySelector('.tm-search input').maxLength).toBe(10);
  });

  // 值为空显示
  it('test showIcon noValue', () => {
    const wrapper = mount(TmSearch, {
      propsData: {
        value: '',
      },
    });
    expect(wrapper.find('.tm-search .el-icon-search').exists()).toBe(true);
  });

  // 有值则显示
  it('test showIcon hasValue', () => {
    const wrapper = mount(TmSearch, {
      propsData: {
        value: '内容',
      },
    });
    expect(wrapper.find('.tm-search .el-icon-search').exists()).toBe(false);
  });

  // 不允许输入特殊字符
  it('test allowSpecialChar notAllow', async () => {
    const wrapper = mount(TmSearch, {
      propsData: {
        allowSpecialChar: false,
        value: '@',
      },
    });
    const el = wrapper.vm.$el;
    await wrapper.vm.handleInput('@');

    expect(el.querySelector('.tm-search input').value).toBe('');
  });

  // 允许输入特殊字符
  it('test allowSpecialChar notAllow', () => {
    const wrapper = mount(TmSearch, {
      propsData: {
        allowSpecialChar: true,
        value: '@',
      },
    });
    expect(wrapper.vm.$el.querySelector('.tm-search input').value).toBe('@');
  });
})
