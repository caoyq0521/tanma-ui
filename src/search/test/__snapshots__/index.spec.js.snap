// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tmSearch render tmSearch 1`] = `
<div class="tm-search el-input el-input--medium el-input--suffix">
  <!----><input type="text" autocomplete="off" placeholder="请输入内容" class="el-input__inner">
  <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="tm-search-icon tm-icon-sousuo"></i><!----><!----><!----><!----></span>
  <!----></span>
  <!---->
  <!---->
</div>
`;
